SteamKit2/SteamKit2/Networking/Steam3/UdpConnection.cs

```diff
@@ -356,8 +365,19 @@ private bool DispatchMessage()
         /// <summary>
         /// Processes incoming packets, maintains connection consistency, and oversees outgoing packets.
         /// </summary>
-        private void NetLoop()
+        private void NetLoop(object param)
         {
+            if (param is IPEndPoint)
+            {
+                remoteEndPoint = param as IPEndPoint;
+            }
+            else
+            {
+                var epTask = param as Task<IPEndPoint>;
+                epTask.Wait();

```

Comment:
Technically unneccesary, as `.Result` will block if the task is not yet completed.